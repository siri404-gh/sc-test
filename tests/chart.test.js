import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import cones from '../cones.json';

jest.mock('chart.js', () => ({
    Chart: jest.fn(),
  }))
  

import Chart from '../src/Chart';

describe('Chart', () => {
    let container;
    const props = {
        cones,
        riskLevel: 10,
        initialSum: 1000,
    };

    beforeEach(() => {
        container = document.createElement('section');
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Should render without props', () => {
        act(() => {
            render(<Chart />, container);
        });
        expect(container).toMatchSnapshot();
    });

    it('Should render with props', () => {
        act(() => {
            render(<Chart {...props} />, container);
        });
        expect(container).toMatchSnapshot();
    });
});