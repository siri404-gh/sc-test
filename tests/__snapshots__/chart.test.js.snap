// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Should render with props 1`] = `
<section>
  <canvas
    height="400"
    id="canvas"
    width="600"
  />
</section>
`;

exports[`Chart Should render without props 1`] = `<section />`;
